Fuzzlyn/Methods/FuncBodyGenerator.cs

```diff
@@ -325,13 +335,86 @@ private StatementSyntax GenIf()
         return gen;
     }
 
+    private StatementSyntax GenThrow()
+    {
+        Debug.Assert(_tryCatchCount > 0);
+
+        return
+            ThrowStatement(
+                ObjectCreationExpression(
+                    QualifiedName(
+                        IdentifierName("System"),
+                        IdentifierName("Exception")))
+                .WithArgumentList(
+                    ArgumentList()));
+    }
+
+    private StatementSyntax GenTryCatch()
+    {
+        int numStatements = Options.BlockStatementCountDist.Sample(_random.Rng);
+        int tryStatements = _random.Next(numStatements);
+        int catchStatements = numStatements - tryStatements;
+        bool doCatchWhen = _random.FlipCoin(0.2);
+
+        // Don't allow 'throw' inside 'try' with a 'catch when' clause; we don't know that it will be caught.
+        if (!doCatchWhen)
+            _tryCatchCount++;
+        BlockSyntax body = GenBlock(numStatements: tryStatements);
+        if (!doCatchWhen)
+            _tryCatchCount--;
+
+        _catchCount++;
+        BlockSyntax catchBody = GenBlock(numStatements: catchStatements);
+        _catchCount--;
+
+        if (doCatchWhen)
+        {
+            // Make it a "catch when" clause
+
+            ExpressionSyntax whenExpression = GenExpression(new PrimitiveType(SyntaxKind.BoolKeyword));
+
+            return
+                TryStatement(
+                    body,
+                    SingletonList<CatchClauseSyntax>(
+                        CatchClause()
+                        .WithDeclaration(
+                            CatchDeclaration(
+                                QualifiedName(
+                                    IdentifierName("System"),
+                                    IdentifierName("Exception"))))
+                        .WithFilter(
+                            CatchFilterClause(whenExpression))
+                        .WithBlock(catchBody)),
+                    null); // no 'finally'
+        }
+        else
+        {
+            return
+                TryStatement(
+                    body,
+                    SingletonList<CatchClauseSyntax>(
+                        CatchClause()
+                        .WithDeclaration(
+                            CatchDeclaration(
+                                QualifiedName(
+                                    IdentifierName("System"),
+                                    IdentifierName("Exception"))))
+                        .WithBlock(catchBody)),
+                    null); // no 'finally'
+
+        }
+    }
+
     private StatementSyntax GenTryFinally()
     {
         int numStatements = Options.BlockStatementCountDist.Sample(_random.Rng);
         int tryStatements = _random.Next(numStatements);
         int finallyStatements = numStatements - tryStatements;
 
+        _tryFinallyCount++;
         BlockSyntax body = GenBlock(numStatements: tryStatements);
+        _tryFinallyCount--;
```

Comment:
Is this used?

Reply:
Looks like I added it but didn't end up using it.